### PR TITLE
Blog app, create data model

### PR DESCRIPTION
**Pull request by firstpj**
In this task, I accomplished the following steps:

1. Developed a new Rails application and assigned it a meaningful name, "blog-app."
2. Initialized the project using Git version control.
3. Ensured that the project included a Postgres database setup after creating the Rails application.
4. Constructed the project schema using the following methods:
   a. Created and executed necessary migration files.
   b. Ensured that table and column names align with the Entity-Relationship Diagram (ERD) provided. Note that the "photo" column in the "users" table should be a hyperlink to an image.
   c. Included foreign keys in the schema. Specifically, in the "posts" table, the column serving as the foreign key linking to the "users" table was named "author_id."
   d. Established corresponding indexes for all columns functioning as foreign keys.
5. Utilized the "sneak peek" section as a reference for any queries related to table or column names or data relationships.

### CODE REVIEW

**Review by roshan-bajgain — APPROVED** [review]

**Review by roshan-bajgain — COMMENTED** [review]
Hi @firstpj, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that the linter is not checked. Please make sure to run the linter and submit the project again.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by roshan-bajgain** [comment]
Hi @firstpj, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that the linter is not checked. Please make sure to run the linter and submit the project again. Please create [.rubocop.yml] and follow the instruction as the link is here https://github.com/microverseinc/linters-config/tree/master/ror

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by topeogunleye** [comment]
Hi @firstpj, Can you please try closing your pull request, then re-opening it to trigger the linters.

or 

 please follow these steps:

1. Edit the file and modify line number 3, replacing `on: pull_request` with `on: push`.
2. Commit and push this change to your repository.
3. Once the linters have executed successfully on the push event, revert line 3 from `on: push` back to `on: pull_request`.
4. Commit and push this final change to restore the original trigger event. 

This sequence ensures that your linters are triggered by the appropriate events during your workflow.

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: Linters
 
-on: pull_request
+on: pull
```
Comment (topeogunleye) [inline .github/workflows/linters.yml:3]: on: push

**Comment by topeogunleye** [comment]
Hi @firstpj, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️

   - You have submitted a project that has failing linters, Kindly make sure your linters are all passing before submitting the PR.

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

>Invalid Code Review Request **does not count** into the code reviews limit.

**Review by joel-kalema — COMMENTED** [review]
Hi @firstpj 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request  🛑 

pirates-of-the-caribbean-johnny-depp


- [ ] You have submitted a project that has linters that do not pass, make sure you solve them and then ask again for a review


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @joel-kalema in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: config/environments/production.rb [hunk 2]
```diff
@@ -1,4 +1,4 @@
-require "active_support/core_ext/integer/time"
+require 'active_support/core_ext/integer/time'
 
 Rails.application.configure do
```
Comment (KayLemba) [inline config/environments/production.rb:3]: While you did a good job working on the project there is an important aspect you missed out in your database yml file. You are using the wrong database adapter please take a look at the screenshot below:
Screenshot 2023-11-04 at 12 11 10 PM

you used sqlite3 as your adapter the correct adapter to use is this:
default: &default
adapter: postgresql
encoding: unicode

...

**Review by KayLemba — CHANGES_REQUESTED** [review]
Hi @firstpj    👋
Good job so far working on the project requirements!
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight 🎉
- No linter errors. ✔️
- Git flow is followed. ✔️
- migrations correctly set up

Required Changes ♻️
Check the comments under the review in the inline comments.

Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AbayomiOlaoye — APPROVED** [review]
## STATUS: APPROVED :green_circle:  

Hi @firstpj  :wave: :wave:, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

applause

## Highlights
- Cleanly written code :white_check_mark: 
- PostgreSQL adapter was used ✅ 
- Nicely implemented models ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [ ] Looking at your commit messages, I can see that you took time to really write commits but I believe you can make them more descriptive and also improve their readability or correctness by capitalizing the initial first letters.

image

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**